api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_online.py

```diff
@@ -40,22 +41,38 @@ def update_single_batch(self, var_vec: npt.NDArray[np.int64], val_vec: npt.NDArr
     def finalize(
         self,
     ) -> Tuple[npt.NDArray[np.float64], npt.NDArray[np.float64], npt.NDArray[np.float64], npt.NDArray[np.float64]]:
-        # TODO: do we want to add ddof param?  Currently hard-wired to 1 (see var calcs)
-
         # correct each batch to account for sparsity
         _mbomv_sparse_correct_batches(self.n_batches, self.n_samples, self.n, self.u, self.M2)
 
         # compute u, var for each batch
         batches_u = self.u
-        batches_var = (self.M2.T / (self.n_samples - 1)).T
+
+        # Note: if N-ddof is less than or equal to 0, we will return Inf - this is consistent
+        # with the numpy.var behavior.
+        batches_var = (self.M2.T / np.maximum(np.zeros(self.n_samples.size), (self.n_samples - self.ddof))).T
 
         # accum all batches using Chan's
         all_u, all_M2 = _mbomv_combine_batches(self.n_batches, self.n_samples, self.u, self.M2)
-        all_var = all_M2 / max(1, (self.n_samples.sum() - 1))
+        all_var = all_M2 / max(0, (self.n_samples.sum() - self.ddof))
```

Comment:
why python `max` and not numpy?

Reply:
Also, given that this will (intentionally) divide by zero, shouldn't this be wrapped in a numpy error context?

Reply:
`max` was already used, I guess because the denominator is just a scalar here?

For the division by zero - it's a good point, I will add it.

Reply:
Re: max - you are right, it isn't a vector operation. Sorry - I misread the code.  OK to leave as is - I don't think it matters at all.

Reply:
On a second thought, I believe we don't need to add an additional error context since it will inherit the one from the division by zero (`np.var` has the same behavior - raises a division by zero warning). What we could do is to add a specific warning that states that ddof must be >= N. Is that desirable?

Reply:
Not sure I understand the point about inheriting an existing numpy error context (where is that defined?)   And I don't think we should add warnings as the behavior is well specified.

What I meant was that the code should look like:
```
with np.errstate(divide='ignore'):
    all_var = all_M2 / max(0, (self.n_samples.sum() - self.ddof))
```